src/utils/translation.cpp

```diff
@@ -358,11 +359,34 @@ Translations::Translations() //: m_dictionary_manager("UTF-16")
 
 // ----------------------------------------------------------------------------
 
+Translations::~Translations()
+{
+    // Remove cached strings
+    for (std::vector<std::pair<wchar_t*, wchar_t*> >::iterator it =
+         m_fribidized_strings.begin(); it != m_fribidized_strings.end(); it++)
+    {
+        delete[] it->first;
+        delete[] it->second;
+    }
+    m_fribidized_strings.clear();
+}   // ~Translations
+
+// ----------------------------------------------------------------------------
+
 const wchar_t* Translations::fribidize(const wchar_t* in_ptr)
 {
 #if ENABLE_BIDI
     if(this->isRTLLanguage())
     {
+        // Test if this string was already fribidized
+        for (std::vector<std::pair<wchar_t*, wchar_t*> >::iterator it =
+             m_fribidized_strings.begin(); it != m_fribidized_strings.end(); it++)
+        {
+            if (wcscmp(it->first, in_ptr) == 0)
+                return it->second;
+        }
```

Comment:
One thing I don't like is that is that on everytime we get a string, this will perform a linear aka O(n) search, performing a string comparison on every step. While probably not catastrophic, it's not very good performance wise

Also I'm not sure I understand why using variadic templates would cause problems with fribidi?